versionhistory.md

```diff
@@ -32,47 +38,28 @@ Status codes:
 * 200 OK - if the version history exists and is accessible by the current user
 * 401 Unauthorized - if you are not authenticated and versioning is not public
 * 403 Forbidden - if you are not logged in with sufficient permissions and versioning is not public
-* 404 Not found - if the version history doesn't exist
+* 404 Not found - if you have the permission to review the version history and it doesn't exist
 
 ## Linked entities
 
-### Versions
+### Oldest Version
 
-**GET /api/versioning/versionhistories/<:versionHistoryId>/versions**
+**GET /api/versioning/versionhistories/<:versionHistoryId>/oldestversion**
 
-Retrieve a pageable list of versions for the provided version history identifier.  
-The versions are ordered by version number descending.
+Retrieve the oldest version in the history 
 
-```json
-{
-  "_embedded": {
-    "versions": [
-      {
-          "id": "102",
-          "version": "2",
-          "type": "version",
-          "created": "2019-10-31T09:44:46.617",
-          "summary": "Author order"
-      },
-      {
-        "id": "101",
-        "version": "1",
-        "type": "version",
-        "created": "2015-11-03T09:44:46.617",
-        "summary": "Fixing some typos in the abstract"
-      }
-    ],
-    "_links": {
-     "self": {
-       "href": "https://dspace7.4science.it/dspace-spring-rest/api/versioning/versionhistories/1/versions"
-     }
-    },
-    "page": {
-     "size": 20,
-     "totalElements": 2,
-     "totalPages": 1,
-     "number": 0
-    }
-  }
-}
-```
+### Current Version
+
+**GET /api/versioning/versionhistories/<:versionHistoryId>/currentversion**
+
+Retrieve the most recent archived version in the history
+
+### Last Version
+
+**GET /api/versioning/versionhistories/<:versionHistoryId>/lastversion**
```

Comment:
I think this should be called `draftversion` if it is only meant to return a Version in the Workspace or Workflow.  This would imply though that it'd return a 404 if no draft is in progress.   This would distinguish it from `currentversion` though, as it's an in-progress or draft version, only if one exists for this item.

If this endpoint is meant to return the same version as `currentversion` in some scenarios, then maybe we should call it `newestversion` and better describe the scenarios where `currentversion` and `newestversion` may (or may not) be the same.

Reply:
yes our idea was that if no draft version exist `currentversion` will be the same than what we named `lastversion`.
So, I'm ok with both options
-  renaming `lastversion` in `newestversion`,  it will be never null
- use `draftVersion` and link it to the workspace or workflow item as appropriate (please note that if it doesn't exist the endpoint need to return 204 no content as it is an association endpoint)

The `draftVersion` would be probably more useful as it will be easier for the client to create a link to resume the current edit if applicable. Access will be the same than the corresponding object so that an administrator or a researcher  (if the flag to allow the original submitter to create new version is on) can get the link to resume the submission from the current item ("you have a pending new version").